### DISCUSSION TITLE
Installing spacy v3 for both env with gpu (server) and without (local)

### DISCUSSION THREAD

**Discussion by salman1993** [explosion/spaCy#6995] [Help: Coding & Implementations]
Currently, in my local Macbook I do `pip install spacy[transformers]` whereas in the server I have to do `spacy[transformers,cuda102]`

Is there anyway to set this up so that the requirements.txt file is the same for both environment and spacy can automatically determine if it should install `cuda102`, `cuda101` or not at all if there is no GPU? I wanna share the same requirements file in my codebase and wondering what's the best way to achieve this.

What's the best practice - should I pin spacy and spacy-transformers separately in requirements.txt? Is there a separate library for the cuda option?

**Comment by salman1993** [comment]
I wanted to follow up on this question. Currently, I have the following lines in my requirements.txt that is deployed to a GPU machine but `spacy.prefer_gpu()` returns False. I have checked that Cuda driver is 10.2 and `torch.cuda.is_available()` is able to find the GPU properly.

```
spacy==3.0.5
spacy-transformers==1.0.1
cupy-cuda102==8.5.0
```

How can I install packages from requirements.txt file so that spacy runs on GPU?

**Reply by polm** [reply]
This is some of the code that spaCy runs to determine the value of `prefer_gpu`. Can you run this without an error?

```
import cupy
import cupyx
import cupy.cuda
from cupy.cuda.compiler import compile_with_cache
 ```
